src/Simplex/Messaging/Agent/Store.hs

```diff
@@ -109,8 +107,13 @@ class Monad m => MonadAgentStore s m where
   removeSndAuth :: s -> ConnAlias -> m ()
   setRcvQueueStatus :: s -> ReceiveQueue -> QueueStatus -> m ()
   setSndQueueStatus :: s -> SendQueue -> QueueStatus -> m ()
+  -- ? make data kind out of AMessage so that we can limit parameter to AMessage A_MSG?
+  -- ? or just throw error / silently ignore other AMessage types?
```

Comment:
It actually seems like AMessage is the wrong type because it does not include the fields present in MSG constructor of ACommand - all the timestamps and IDs...

AMessage is something that we parse out of the SMP.MSG + additional field, but it may be that we just need a separate type that we save to the database that could be used to create agent MSG.

It seems like MessageDelivery is the type that should be saved to the database - it missed some fields but it is closest to what we need.